zebra-utils/src/bin/block-template-to-proposal/args.rs

```diff
@@ -0,0 +1,66 @@
+//! block-template-to-proposal arguments
+//!
+//! For usage please refer to the program help: `block-template-to-proposal --help`
+
+use std::str::FromStr;
+
+use structopt::StructOpt;
+
+use zebra_chain::{serialization::DateTime32, BoxError};
+
+/// The source of the time in the block proposal header.
+#[derive(Clone, Debug, Eq, PartialEq)]
+pub enum TimeSource {
+    /// The `curtime` field in the template.
+    CurTime,
+
+    /// The `mintime` field in the template.
+    MinTime,
+
+    /// The `maxtime` field in the template.
+    MaxTime,
+
+    /// The supplied time, clamped within the template's `[mintime, maxtime]`.
+    Clamped(DateTime32),
+
+    /// The raw supplied time, ignoring the `mintime` and `maxtime` in the template.
+    ///
+    /// Warning: this can create an invalid block proposal.
+    Raw(DateTime32),
+}
```

Comment:
@arya2 you might be interested in this draft code - it shows the different times we could test with.

I'm happy if you want to put parts of it in PR #5884, or leave it for me for tomorrow.

Reply:
I altered `proposal_block_from_template` to return the three times for the acceptance test, but using an enum may be better. I'll leave this for you.